examples/zephyr/certificate_rotation/src/main.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2025 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(cert_rotation, LOG_LEVEL_DBG);
+
+#include <golioth/client.h>
+#include <golioth/pki.h>
+#include <samples/common/net_connect.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/coap.h>
+#include <zephyr/device.h>
+#include <zephyr/fs/fs.h>
+#include <zephyr/fs/littlefs.h>
+#include <zephyr/sys/reboot.h>
+#include <mbedtls/x509_csr.h>
+
+#define CRT_PATH "/lfs1/credentials/crt.der"
+#define KEY_PATH "/lfs1/credentials/key.der"
+
+static const uint8_t tls_ca_crt[] = {
+#include "golioth-systemclient-ca_crt.inc"
+};
+
+static const uint8_t tls_secondary_ca_crt[] = {
+#if CONFIG_GOLIOTH_SECONDARY_CA_CRT
+#include "golioth-systemclient-secondary_ca_crt.inc"
+#endif
+};
+
+#define CSR_BUF_SIZE 300
+
+struct golioth_client *client;
+static K_SEM_DEFINE(connected, 0, 1);
+
+static struct
+{
+    struct k_sem valid;
+    uint8_t *buf;
+    size_t len;
+} new_cert;
+
+static int psa_rng_for_mbedtls(void *p_rng, unsigned char *output, size_t output_len)
+{
+    return psa_generate_random(output, output_len);
+}
```

Comment:
Slight preference for using `mbedtls_psa_get_random` (`f_rng`) and `MBEDTLS_PSA_RANDOM_STATE` (`p_rng`) from `mbedtls/psa_util.h` rather than providing our own wrapper here.